cryptoki/src/mechanism/mechanism_info.rs

```diff
@@ -228,6 +231,27 @@ impl MechanismInfo {
     pub fn ec_compressed(&self) -> bool {
         self.flags.contains(MechanismInfoFlags::EC_COMPRESS)
     }
+
+    /// True if the mechanism can be used to encrypt messages
+    ///
+    /// See [`Session::encrypt_message`](crate::session::Session::encrypt_message)
+    pub fn message_encrypt(&self) -> bool {
+        self.flags.contains(MechanismInfoFlags::MESSAGE_ENCRYPT)
+    }
+
+    /// True if the mechanism can be used to decrypt encrypted messages
+    ///
+    /// See [`Session::decrypt`](crate::session::Session::decrypt_message)
```

Comment:
The below is important to catch stuff like here (`decrypt` vs `decrypt_message`).